src/lang/terms/holds.pl

```diff
@@ -206,3 +207,64 @@
 data_base_type_(_Property,DataValue,string) :- string(DataValue),!.
 data_base_type_(_Property,true,bool) :- !.
 data_base_type_(_Property,false,bool) :- !.
+
+     /*******************************
+     *	    UNIT TESTS	     		*
+     *******************************/
+
+:- use_module(library('lang/query'),
+  [ tell/1]).
+
+:- use_module(library('lang/terms/temporal')).
+:- use_module(library('db/tripledb_tests')).
+:- use_module(library('db/tripledb'),
+  [ tripledb_load/2 ]).
+
+:- use_module('holds.pl').
+
+:- begin_tripledb_tests(
+    'lang_holds',
+    'package://knowrob/owl/test/swrl.owl',
+    [ namespace('http://knowrob.org/kb/swrl_test#')
+    ]).
+
+
+test('ask triples holds') :-
+  assert_true(holds(test:'Ernest', test:'hasSibling', test:'Fred')).
+
+test('ask holds with arity 1') :-
+  assert_true(holds(test:'hasHeightInMeters'(test:'RectangleBig',13.5))).
+
+test('ask holds using operators >, <, = a value in float') :-
+  assert_true(holds(test:'RectangleBig',test:'hasHeightInMeters', =(13.5))),
+  assert_true(holds(test:'RectangleBig',test:'hasHeightInMeters', <(15.0))),
+  assert_true(holds(test:'RectangleBig',test:'hasHeightInMeters', >(10.5))).
+
+test('ask holds using operators = < > a value in integer') :-
+  assert_true(holds(test:'RectangleSmall',test:'hasHeightInMeters', =(6))),
+  assert_true(holds(test:'RectangleSmall',test:'hasHeightInMeters', =<(9))),
+  assert_true(holds(test:'RectangleSmall',test:'hasHeightInMeters', <(7))),
+  assert_true(holds(test:'RectangleSmall',test:'hasHeightInMeters', >=(5))),
+  assert_true(holds(test:'RectangleSmall',test:'hasHeightInMeters', >(3))).
+
+test('tell Lea hasNumber') :-
+  assert_true(tell(holds(test:'Lea', test:'hasNumber', '+493564754647'))),
+  assert_true(holds(test:'Lea', test:'hasNumber', '+493564754647')).
+
+test('tell the rectangle size during a time interval') :-
+  tell(holds(test:'RectangleBig',test:'hasHeightInMeters', 15.2) during [25,45]).
```

Comment:
better would be a testcase where the size actually changed at some time, so that you can test that holds captures this change

Reply:
also a negative example would be good (assert_false)